### PR TITLE
fix(network): throw errors from txQueue calls

### LINKED ISSUE #315: Gas estimation errors do not propagate up to client
When a transaction, called in the client by
```
systems["system.xxx"].executeTyped(...)
```
cause a revert, an error is thrown:

Screenshot 2022-12-16 at 16 56 29

https://github.com/latticexyz/mud/blob/915506d7e7ca0b5a68afb646388bb9d4bb689879/packages/network/src/createTxQueue.ts#L192

What I would like is to be able to catch this error on the app-level.

### PR DESCRIPTION
**Pull request by holic**
closes #315

### CODE REVIEW

File: packages/network/src/createTxQueue.ts [hunk 1]
```diff
@@ -190,7 +190,9 @@ export function createTxQueue<C extends Contracts>(
         gasLimit = await txRequest.estimateGas();
       } catch (e) {
         console.error("[TXQueue] GAS ESTIMATION ERROR", e);
-        return txRequest.cancel();
+        throw e;
+      } finally {
+        txRequest.cancel();
```
Comment (Kooshaba) [inline packages/network/src/createTxQueue.ts:195]: this cancels every tx

Comment (holic) [reply packages/network/src/createTxQueue.ts:195]: ahh I see, so it makes sure that if one gas estimate fails, the whole queue is canceled?

Comment (holic) [reply packages/network/src/createTxQueue.ts:195]: https://github.com/latticexyz/mud/blob/f18c39831f5d5ab0186529eb2cfaee4991570e6f/packages/network/src/createTxQueue.ts#L155

I see now that this just throws a generic error. Maybe I should rip out `txRequest.cancel` and just let the real error bubble up.

Comment (holic) [reply packages/network/src/createTxQueue.ts:195]: > this cancels every tx

I see what you mean now 🙈

File: packages/network/src/createTxQueue.ts [hunk 2]
```diff
@@ -219,6 +224,8 @@ export function createTxQueue<C extends Contracts>(
         // Nonce handeling
         if (shouldIncreaseNonce) incNonce();
         if (shouldResetNonce) await resetNonce();
+        // Bubble up error
+        if (error) txRequest.cancel(error);
```
Comment (holic) [inline packages/network/src/createTxQueue.ts:228]: This still doesn't help with tx errors if it passes the gas estimation and returns a plain "transaction failed" error message:
image

**Review by alvrs — APPROVED** [review]
lgtm 🚢

**Pull request merged**